src/AutoMapper.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -110,7 +115,25 @@ void ConfigAction(IServiceProvider serviceProvider, IMapperConfigurationExpressi
             }
 
             services.AddSingleton<IConfigurationProvider>(sp => new MapperConfiguration(cfg => ConfigAction(sp, cfg)));
-            return services.AddScoped<IMapper>(sp => new Mapper(sp.GetRequiredService<IConfigurationProvider>(), sp.GetService));
+            switch (serviceLifetime)

```

Comment:
There is no need to replicate code here. There is an Add method that does the job.

Reply:
Not clear on what you're suggesting here.

Reply:
services.Add(...) instead of the switch.

Reply:
```c#
var serviceDescriptor = new ServiceDescriptor(
    typeof(IMapper), 
    sp => new Mapper(sp.GetRequiredService<IConfigurationProvider>(), sp.GetService),
    serviceLifetime);
services.TryAdd(serviceDescriptor);
```